contrib/globalsign/mgo/mgo.go

```diff
@@ -0,0 +1,335 @@
+// Package mgo provides functions and types which allow tracing of the MGO MongoDB client (https://github.com/globalsign/mgo)
+package mgo // import "gopkg.in/DataDog/dd-trace-go.v1/contrib/globalsign/mgo"
+
+import (
+	"github.com/globalsign/mgo"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// Dial opens a connection to a MongoDB server and ties it to the
+// given context for tracing MongoDB calls.
+func Dial(url string, opts ...MongoOption) (*Session, error) {
+	session, err := mgo.Dial(url)
+	s := &Session{
+		Session: session,
+	}
+
+	defaults(&s.cfg)
+	for _, fn := range opts {
+		fn(&s.cfg)
+	}
+
+	return s, err
+}
+
+// Session wraps a Session with the Context and Config information
+// needed to support tracing.
+type Session struct {
+	*mgo.Session
+
+	cfg mongoConfig
+}
+
+func newChildSpanFromContext(config mongoConfig) ddtrace.Span {
+	name := "mongodb.query"
+	span, _ := tracer.StartSpanFromContext(
+		config.ctx,
+		name,
+		tracer.SpanType(ext.SpanTypeMongoDB),
+		tracer.ServiceName(config.serviceName),
+		tracer.ResourceName("mongodb.query"))
+
+	return span
+}
+
+// Run invokes and traces Session.Run
+func (s *Session) Run(cmd interface{}, result interface{}) (err error) {
+	span := newChildSpanFromContext(s.cfg)
+	err = s.Session.Run(cmd, result)
+	span.Finish(tracer.WithError(err))
+	return
+}
+
+// Database wraps a Database with the Context and Config information
```

Comment:
Let's not use `Context` and `Config` here either. It's unclear what they are. Let's refer strictly to this structure without its context use.

I'd be fine using the same text as `Session`. I find it's a good and accurate (surprise free) description: it states that it embeds the original structure and also mentions specifically that it traces only attached methods. We've had issues in the past where users were calling embedded methods because we weren't wrapping them and not understanding why tracing has gone missing. It can be tricky and having clear documentation helps us avoid tickets.